recipes/cgal/all/conanfile.py

```diff
@@ -86,23 +63,13 @@ def package(self):
         self.copy("LICENSE*", dst="licenses", src=self._source_subfolder)
         cmake = self._configure_cmake()
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        if self.options.get_safe("shared"):
-            for root, _, filenames in os.walk(os.path.join(self.package_folder, "bin")):
-                for filename in filenames:
-                    if not filename.endswith(".dll"):
-                        os.unlink(os.path.join(root, filename))
-        else:
-            tools.rmdir(os.path.join(self.package_folder, "bin"))
+        files.rmdir(self, os.path.join(self.package_folder, "share"))
+        files.rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        files.rmdir(self, os.path.join(self.package_folder, "bin"))
 
     def package_info(self):
         # TODO: add components
         self.cpp_info.names["cmake_find_package"] = "CGAL"
         self.cpp_info.names["cmake_find_package_multi"] = "CGAL"
-        if not self.options.header_only:
-            self.cpp_info.libs = tools.collect_libs(self)
-        if self.settings.os == "Linux" and (self.options.with_cgal_core or self.options.with_cgal_imageio):
-            self.cpp_info.system_libs.append("m")
-        if not self.options.header_only:
-            self.cpp_info.defines.append("CGAL_NOT_HEADER_ONLY=1")
+        self.cpp_info.system_libs.append("m")
```

Comment:
```suggestion
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs.append("m")
```
libm is not needed for every OS.